src/App.tsx

```diff
@@ -15,71 +16,95 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+interface State {
+  selectedProduct: string | null,
+}
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    selectedProduct: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleSelectProduct = (product: string) => {
+    this.setState({
+      selectedProduct: product,
+    });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  handleRemoveProduct = () => {
+    this.setState({
+      selectedProduct: '',
+    });
+  };
+
+  render() {
+    const { selectedProduct } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+    return (
+      <main className="section container">
+        <h1 className="title">
+          {selectedProduct
+            ? `${selectedProduct} is selected`
+            : 'No goods selected'}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+          {selectedProduct && (
+          /* eslint-disable-next-line jsx-a11y/control-has-associated-label */
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              className="delete ml-3"
+              onClick={this.handleRemoveProduct}
+            />
+          )}
+        </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {products.map(product => {
+              const isSelected = product === selectedProduct;
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+              return (
+                <tr
+                  data-cy="Good"
+                  key={product}
+                  className={classNames(
+                    {
+                      'has-background-success-light': isSelected,
+                    },
+                  )}
```

Comment:
let's have it more concise
```suggestion
                  className={classNames({
                      'has-background-success-light': isSelected,
                    })}
```